dockerfiles/diy/build_docker_image.sh

```diff
@@ -1,5 +1,7 @@
 #! /bin/sh
 
+cd "$(dirname "$0")"
```

Comment:
It's a good idea, but it's probably neater to create a subshell for the bits where we are doing the relative copies. It would avoid `cd` and being in an unexpected directory if your arg parsing failed or whatever.

I could probably restructure this to be neater altogether, but I don't think it's widely used.

Anyway, something like this at line 149:
```
# Create a subshell to avoid having to cd back
(
cd "$(dirname "${0}")" || terminate "Couldn't cd to source location";

# Copy in our dockerfiles, we'll need them later
mkdir "${BUILD_DIR}/dockerfiles"
cp dockerfiles/Dockerfile.repo "${BUILD_DIR}" || terminate "Couldn't copy dockerfiles to ${BUILD_DIR}"
cp ../Dockerfile.router "${BUILD_DIR}" || terminate "Couldn't copy dockerfiles to ${BUILD_DIR}"
cp ../router.yaml "${BUILD_DIR}/dockerfiles" || terminate "Couldn't copy ../router.yaml to ${BUILD_DIR}"
)
```

Reply:
That's fair and I can make that change, although I'll note that immediately after the part you mention there's an unconditional `cd` anyway!

Reply:
I moved it here but without the subshell due to the other cd. But happy to go with your original suggestion if you want.